src/mlpack/core/dists/discrete_distribution.hpp

```diff
@@ -219,6 +244,25 @@ class DiscreteDistribution
   std::vector<arma::vec> probabilities;
 };
 
+/**
+ * Calculates the Discrete log probability function for each
+ * data point (column) in the given matrix
+ *
+ * @param x List of observations.
+ * @param probabilities Output log probabilities for each input observation.
+ */
+inline void DiscreteDistribution::LogProbability(
+    const arma::mat& x,
+    arma::vec& logProbabilities) const
+{
+  arma::vec logProbs(x.n_cols);
+  for (size_t i = 0; i < x.n_cols; i++)
+    logProbs(i) = log(Probability(x.unsafe_col(i)));
+
+  logProbabilities = logProbs;
+}
+
+
```

Comment:
Please remove the extraneous newline here.

Reply:
Sure. Thanks.